docs/README

```diff
@@ -147,6 +147,26 @@ meaning that it is possible to unmap any aligned area and turn it into a hole.
 This is implemented using the "FALLOC_FL_PUNCH_HOLE" "mode" of the
 "fallocate()" system call.
 
+Note: If running on the ZFS file system, the default module parameters
+configuration can cause the finding of mapped and unmapped areas to fail.
+This can be fixed temporarily by doing the following:
+
+$ echo 1 | sudo tee -a /sys/module/zfs/parameters/zfs_dmu_offset_next_sync
+
+However, if a permanent solution is required then perform the following:
+
+$ echo "options zfs zfs_dmu_offset_next_sync=1" | sudo tee -a /etc/modprobe.d/zfs.conf
+
+Depending upon your Linux distro, you may also need to do the following to 
+ensure that the permanent change is updated in all your initramfs images:
+
+$ sudo update-initramfs -u -k all
+
+To verify the temporary or permanent change has worked you can use the following
+which should return "1":
+
+$ cat /sys/module/zfs/parameters/zfs_dmu_offset_next_sync
```

Comment:
I wonder, how hard it is to do this run-time: verify that we are dealing with ZFS and just check that file? Would you be enthusiastic enough implementing such a check?

Reply:
should be doable, the only complexity here is to work out if the files being operated on reside on a ZFS file system or not.

Reply:
I'll spin this off into a separate issue ticket